app/assets/javascripts/govuk_publishing_components/components/modal-dialogue.js

```diff
@@ -0,0 +1,92 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function ModalDialogue () { }
+
+  ModalDialogue.prototype.start = function ($module) {
+    this.$module = $module[0]
+    this.$dialogBox = this.$module.querySelector('.gem-c-modal-dialogue__box')
+    this.$closeButton = this.$module.querySelector('.gem-c-modal-dialogue__close-button')
+    this.$body = document.querySelector('body')
+
+    this.$module.open = this.handleOpen.bind(this)
+    this.$module.close = this.handleClose.bind(this)
+    this.$module.focusDialog = this.handleFocusDialog.bind(this)
+    this.$module.boundKeyDown = this.handleKeyDown.bind(this)
+
+    var $triggerElement = document.querySelector(
+      '[data-toggle="modal"][data-target="' + this.$module.id + '"]'
+    )
+
+    if ($triggerElement) {
+      $triggerElement.addEventListener('click', this.$module.open)
+    }
+
+    if (this.$closeButton) {
+      this.$closeButton.addEventListener('click', this.$module.close)
+    }
```

Comment:
Much of this seems to have been written intentionally without making use of jquery, is there a reason for that?  It seems like lines 18 to 28 could be simplified if we didn't.

Also do we need to check for the existence of `$triggerElement` and `$closeButton`? Is there a scenario in which they don't exist? And even if they didn't, if we tried to attach an eventListener to them and the code can't find them, would that cause an error, or would it fail silently? (I'm genuinely not sure of the answer to that last question, so this is me wanting to find out)

Reply:
Short-term I'd would like not to enforce jQuery as a dependency for projects that are do not need it. Long term I'd like to question the need of it in GOV.UK.